src/text-error-reporter.cpp

```diff
@@ -23,16 +35,42 @@ void text_error_reporter::set_source(padded_string_view input,
 void text_error_reporter::report_impl(error_type type, void *error) {
   QLJS_ASSERT(this->file_path_);
   QLJS_ASSERT(this->locator_.has_value());
+
+  bool format_escape_errors = false;
+
+  switch (this->escape_errors_) {
+  case escape_errors::auto_:
+#if defined(_WIN32)
+    if (this->output_.rdbuf() == std::cerr.rdbuf() && _isatty(_fileno(stderr)))
+#else
+    if (this->output_.rdbuf() == std::cerr.rdbuf() && isatty(fileno(stderr)))
```

Comment:
Nit: No need for `fileno`:
```suggestion
    if (this->output_.rdbuf() == std::cerr.rdbuf() && isatty(STDIN_FILENO))
```